addons/sup_combatsupport/scripts/NEO_radio/functions/misc/fn_RespawnArtyAsset.sqf

```diff
@@ -41,22 +33,18 @@ if (ARTY_RESPAWN_LIMIT == 0) exitwith {
 sleep _respawn;
 ARTY_RESPAWN_LIMIT = ARTY_RESPAWN_LIMIT - 1;
 
-//Remove from all side-lists
-_toDelete = [];
+//Validate rounds
 {
-        private ["_sideArray","_sideIn"];
-        _sideIn = _x;
-        _sideArray = NEO_radioLogic getVariable [format["NEO_radioArtyArray_%1", _sideIn],[]];
-        {
-            if (count _units > 0) then {
-                _toDelete pushback _x;
-
-                _sideArray set [_foreachIndex, -1];
-                _sideArray = _sideArray - [-1];
-            };
-        } foreach _sideArray;
-        NEO_radioLogic setVariable [format["NEO_radioArtyArray_%1", _sideIn], _sideArray, true];
-} foreach _sides;
+    if ((_x select 0) in _roundsUnit) then
+    {
+        _roundsAvailable pushback _x;
+    };
+} forEach _rounds;
+
+//This unit cannot be used anymore, remove from side-list
+_sideArray = NEO_radioLogic getVariable [format["NEO_radioArtyArray_%1", _side], []];
+_sideArray = _sideArray - [[_leader, _grp, _callsign, _units, _roundsAvailable]];
```

Comment:
Removing by index seems a whole lot "safer". Like find the index by comparing `_leader` against the leader in the array or something.